src/biogeophys/test/Irrigation_test/test_irrigation.pf

```diff
@@ -645,26 +630,26 @@ contains
     class(TestIrrigation), intent(inout) :: this
     real(r8), allocatable :: deficits(:,:)
     real(r8) :: expected
-    integer :: time_prev_orig
     integer :: time
     integer :: expected_num_time_steps
 
     ! Setup
     call this%setupSinglePatch()
     call this%setupIrrigation()
-    time_prev_orig = this%time_prev
 
     ! Call irrigation routines for long enough that irrigation should go to 0
     expected_num_time_steps = this%irrigation_params%irrig_length / dtime
     do time = 1, expected_num_time_steps + 1
        call this%calculateAndApplyIrrigation()
-       this%time_prev = this%time_prev + dtime
+       call advance_timestep()
     end do
     ! The following assertion is mainly here to make sure the test is working as intended
     @assertEqual(0._r8, this%waterflux%qflx_irrig_patch(bounds%begp))
 
     ! Now reset time, change soil moisture, and make sure that irrigation happens as expected
-    this%time_prev = time_prev_orig
+    call unittest_timemgr_teardown()
+    call unittest_timemgr_setup(dtime=dtime)
```

Comment:
I don't think you need to teardown and re-setup the time manager here: you should be able to simply reset the time. (If necessary, you could use the same time but the following day.)

Reply:
Yep, you are right.